feature: Add Code snippets for the Qomponents

andyogaga opened the pull request:
This PR does the following:

- Adds new Qomponents documentations.
- Updates tailwind to v4
- Updates @seeqdev/qomponents to v153
- Adds the type definitions of all qomponents
- Adds a code snippet for each qomponent.

andyogaga committed:
feat: Add QTip component and documentation

feat: Enhance Select component with detailed props documentation

feat: Update Tabs component to include props and type definitions

feat: Expand TextArea component with comprehensive props documentation

feat: Revamp TextField component to include extensive props details

feat: Improve ToolbarButton component with detailed props documentation

refactor: Remove Tooltip component as it is no longer needed

chore: Delete unused Tailwind CSS configuration and styles

chore: Integrate Tailwind CSS with Vite configuration

andyogaga committed:
Refactor QomponentWrapper to remove view state and update Preview component usage; add new Qomponents: AlertQomponent, ButtonGroupQomponent, CarouselQomponent, CollapseQomponent, GettingStartedQomponent, IconQomponent, InputGroupQomponent, ProgressBarQomponent, SeeqActionDropdownQomponent, SliderQomponent; update QomponentsLayout for improved styling and structure.

andyogaga committed:
feat: Add Seeq logo asset and update GettingStartedQomponent to use local logo

andyogaga committed:
feat: Update Qomponents documentation with detailed type definitions and props for ButtonWithDropdown, SeeqActionDropdown, ProgressBar, InputGroup, and others; remove unnecessary comments and improve styling.

bmartinelle reviewed (commented):
First off: very excited to see this coming together! It will be a great resource for our end-users! 

Here are some thoughts I had looking at this more closely:

- The Code Sandbox examples should include all properties and show how the onChange handlers for each of them are called
- Let’s switch up the order and show the codeSandbox first and then the props 
- Can we make the code sandbox preview a lot smaller? There’s a LOT of whitespace
- Can we add an indicator on the sidePanel so it shows what they are looking at? Some qomponents have a lot of props and it’s easy to forget what I am looking at (Button, ButtonWithDropDown, ButtonWithPopover …)

from a code perspective I think we really have to create a component wrapper for the properties as that's a lot of repeated tags that will make it hard to maintain.
Something like 
```
<Properties>
    <PropertyEntry name={name} type={} required={} description={}/>
    <PropertyEntry name={name} type={} required={} description={}/>
    <PropertyEntry name={name} type={} required={} description={}/>
     ....
</Properties>
```
will be better. 

Please let me know if you have any questions!

andyogaga committed:
Enhance documentation for Qomponents

- Added detailed component descriptions and usage examples for various Qomponents including Accordion, Alert, ButtonGroup, Button, ButtonWithDropdown, ButtonWithPopover, Carousel, Checkbox, Collapse, Getting Started, Icon, InputGroup, Modal, ProgressBar, Qtip, SeeqActionDropdown, Select, Slider, Tabs, TextArea, TextField, and ToolbarButton.
- Updated prop definitions to improve clarity and consistency across components.
- Ensured all components now include comprehensive information on functionality, props, and usage patterns.

andyogaga committed:
Refactor Qomponent Documentation Components

- Introduced `ComponentPropsDisplay` for consistent rendering of component props across various Qomponents.
- Added `ComponentSubPropsDisplay` for displaying sub-properties and type definitions with enhanced styling.
- Updated `InputGroupQomponent`, `ModalQomponent`, `ProgressBarQomponent`, `QtipQomponent`, `SeeqActionDropdownQomponent`, `SelectQomponent`, `SliderQomponent`, `TabsQomponent`, `TextAreaQomponent`, `TextFieldQomponent`, and `ToolbarButtonQomponent` to utilize the new display components for improved documentation structure and readability.
- Removed redundant code and improved overall organization of prop and type definitions in the documentation.

andyogaga committed:
Refactor Qomponents to use generated prop definitions

- Updated ProgressBarQomponent to import type definitions and props from generated files.
- Refactored SeeqActionDropdownQomponent to utilize generated props and type definitions.
- Modified SelectQomponent to use generated select props and type definitions.
- Changed SliderQomponent to import slider props from generated definitions.
- Updated TabsQomponent to use generated tab props.
- Refactored TextAreaQomponent to utilize generated textarea props.
- Changed TextFieldQomponent to import text field props from generated definitions.
- Updated ToolbarButtonQomponent to use generated toolbar button props.

andyogaga committed:
chore: update dependencies and improve type definitions

- Updated @seeqdev/qomponents from ^0.0.156 to 0.0.159 in package.json.
- Changed React and ReactDOM versions from ^19.0.0 to 19.0.0 for consistency.
- Enhanced type safety in App.tsx by specifying the type of the value in the onChange handler.
- Updated import statement in AvailableIcons.tsx to import from the main package instead of dist.
- Improved type definitions and descriptions in AlertProps.ts for clarity and consistency.
- Added new InputGroupProps.ts file with detailed prop definitions for InputGroup component.
- Updated index.ts to export new InputGroupProps and alert inherited props.
- Refactored InputGroupQomponent.tsx to utilize the new InputGroupProps definitions.
- Created a new TypeScript declaration file for @seeqdev/qomponents to define component types.

andyogaga committed:
chore: update @seeqdev/qomponents to version 0.0.160 and remove obsolete type definitions

andyogaga committed:
chore: add Dependabot configuration for automatic updates

andyogaga committed:
fix: update description of Qomponents to include additional components

andyogaga committed:
Test/dependabot-simulation (#5)

* Create dependabot.yml

* trying to simulate package update

* chore: simplify dependabot configuration by removing outdated auto-update workflow

---------

Co-authored-by: Birgit Martinelle <birgit.martinelle@gmail.com>

andyogaga committed:
Merge branch 'main' of github.com:seeq12/qomponents-documentation into feature/ao/add-code-snippet-CRAB-48009

andyogaga committed:
Enhance ButtonQomponent and TextFieldQomponent documentation with type definitions and inherited props display
